js/coinone.js

```diff
@@ -371,78 +371,253 @@ module.exports = class coinone extends Exchange {
     }
 
     parseOrder (order, market = undefined) {
-        //
-        //     {
-        //         "index": "0",
-        //         "orderId": "68665943-1eb5-4e4b-9d76-845fc54f5489",
-        //         "timestamp": "1449037367",
-        //         "price": "444000.0",
-        //         "qty": "0.3456",
-        //         "type": "ask",
-        //         "feeRate": "-0.0015"
-        //     }
-        //
+        // [WARNING] This implementation is only for fetchOrder()
         const info = this.safeValue (order, 'info');
+        const currency = this.safeString (info, 'currency');
+        let symbol = undefined;
+        if (market === undefined) {
+            const marketId = currency.toLowerCase ();
+            if (marketId in this.markets_by_id) {
+                market = this.markets_by_id[marketId];
+            }
+        }
+        if (market !== undefined) {
+            symbol = market['symbol'];
+        }
         const id = this.safeStringUpper (info, 'orderId');
         const timestamp = this.safeTimestamp (info, 'timestamp');
-        const status = this.parseOrderStatus (this.safeString (order, 'status'));
-        let cost = undefined;
+        let feeCurrency = undefined;
         let side = this.safeString (info, 'type');
         if (side.indexOf ('ask') >= 0) {
             side = 'sell';
+            feeCurrency = market['quote'];
         } else {
             side = 'buy';
+            feeCurrency = market['base'];
         }
         const price = this.safeFloat (info, 'price');
-        const amount = this.safeFloat (info, 'qty');
-        const remaining = this.safeFloat (info, 'remainQty');
+        let amount = this.safeFloat (info, 'qty');
+        let remaining = this.safeFloat (info, 'remainQty');
         let filled = undefined;
-        if (amount !== undefined) {
-            if (remaining !== undefined) {
-                filled = amount - remaining;
-            }
-            if (price !== undefined) {
-                cost = price * amount;
-            }
-        }
-        const currency = this.safeString (info, 'currency');
-        const fee = {
-            'currency': currency,
-            'cost': this.safeFloat (info, 'fee'),
-            'rate': this.safeFloat (info, 'feeRate'),
-        };
-        let symbol = undefined;
-        if (market === undefined) {
-            const marketId = currency.toLowerCase ();
-            if (marketId in this.markets_by_id) {
-                market = this.markets_by_id[marketId];
+        let status = undefined;
+        const orderStatus = this.safeString (order, 'status');
+        if (orderStatus === 'live') {
+            status = 'open';
+            filled = 0;
+            if (amount > remaining) {
+                amount = remaining;
```

Comment:
I'm trying to understand the logic behind this assignment, can you elaborate please? If you paste the verbose request+response – that would really help. Thx!

Reply:
I thought this is hard to understand.
The response from Coninone privatePostOrderOrderInfo() is somewhat tricky.
Let me show you several examples.

### Example 1. an open order which was neither filled nor partially-canceled
```json
{
    "errorCode":   "0",
    "status":   "live",
    "info": {
        "orderId": "50ff3088-1e4d-11e9-9ec7-00e04c3600d7",
        "feeRate": "0.002",
        "fee": "0.0",
         "remainQty": "2.7944",
         "timestamp": "1591274321",
         "price": "3627.0",
         "currency": "ATOM",
         "type": "ask",
         "qty": "2.7944"
    },
    "result":   "success"
}
```
This case is straightforward.

### Example 2. an open order which was never filled but partially-canceled
```json
{
    "errorCode":   "0",
    "status":   "live",
    "info": {
        "orderId": "50ff95fd-1e4d-11e9-9ec7-00e04c3600d7",
        "feeRate": "0.002",
        "fee": "0.0",
         "remainQty": "0.4",
         "timestamp": "1591275188",
         "price": "3610.0",
         "currency": "ATOM",
         "type": "bid",
         "qty": "1.0"
    },
    "result":   "success"
}
```
**This is the most problematic case.**
Because "remainQty" is less than "qty", it is natural for us to think this order was partially filled.
But it is not. The status "live" means that this order was never filled.
In fact, this order was partially canceled.

### Example 3. an open order which was partially-filled
```json
{
    "errorCode":   "0",
    "status":   "partially_filled",
    "info": {
        "orderId": "50ff3088-1e4d-11e9-9ec7-00e04c3600d7",
        "feeRate": "0.002",
        "fee": "3.627",
         "remainQty": "2.2944",
         "timestamp": "1591274321",
         "price": "3627.0",
         "currency": "ATOM",
         "type": "ask",
         "qty": "2.7944"
    },
    "result":   "success"
}
```
Like example 2 above, "remainQty" is less than "qty". But this is a different case.
The status "partially_filled" means itself.

### Example 4. a closed order which was ever filled
```json
{
    "errorCode":   "0",
    "status":   "filled",
    "info": {
        "orderId": "50ff3088-1e4d-11e9-9ec7-00e04c3600d7",
        "feeRate": "0.002",
        "fee": "3.627",
         "remainQty": "0.0",
         "timestamp": "1591274344",
         "price": "3627.0",
         "currency": "ATOM",
         "type": "ask",
         "qty": "0.5"
    },
    "result":   "success"
}
```
The status "filled" means this order was closed and ever filled (fully or partially).
"qty" means not the ordered amount but the filled amount.
"remainQty" is always zero for this case.

### Example 5. a closed(canceled) order without no fills
In this case, Coineone API will return 'Not Found'

Reply:
@kroitor Sorry for late. Here is:

```json
{
    "result": "success",
    "errorCode": "0",
    "limitOrders":[
        {
            "index": "0",
            "orderId": "525f314f-1e4d-11e9-9ec7-00e04c3600d7",
            "timestamp": "1591953153",
            "price": "89.27",
            "qty": "57.8274",
            "type": "ask",
            "feeRate": "0.002"
        }
    ]
}
```